website/static/js/fangorn.js

```diff
@@ -642,7 +643,7 @@ function _fangornActionColumn (item, col) {
 
     // Upload button if this is a folder
     // If File and FileRead are not defined dropzone is not supported and neither is uploads
-    if (window.File && window.FileReader && item.kind === 'folder' && item.data.provider && item.data.permissions.edit) {
+    if (window.uploadEnabled && window.File && window.FileReader && item.kind === 'folder' && item.data.provider && item.data.permissions.edit) {
```

Comment:
This doesn't solve the problem I previously mentioned. The issue is that we don't want to rely on `window` variables *that we set* (checking for browser features, e.g. `window.File` , is fine). 

You should be able to check the treebeard object to see if uploads are enabled.